server/core/src/repl/mod.rs

```diff
@@ -450,8 +456,22 @@ async fn repl_task(
 
     info!("Replica task for {} has started.", origin);
 
+    let mut last_working_address: Option<SocketAddr> = None;
+
     // Okay, all the parameters are setup. Now we wait on our interval.
     loop {
+        // if it worked last time, then let's use it this time!
+        // I sure hope this works. - @yaleman
+        let mut sorted_socket_addrs = match last_working_address {
+            Some(addr) => vec![addr],
+            None => vec![],
+        };
+        socket_addrs.iter().cloned().for_each(|addr| {
```

Comment:
I'd throw in a comment that this is an exponential search/insert process, but that's okay because the number of addrs will be small, so it's cheap over all.

Reply:
Done!